ts/annotations/MockPoint.ts

```diff
@@ -0,0 +1,541 @@
+/* *
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+import H from '../parts/Globals.js';
+
+/**
+ * Internal types.
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        interface AnnotationMockLabelOptionsObject {
+            x: (number|null);
+            y: (number|null);
+            point: AnnotationMockPoint;
+        }
+        class AnnotationMockPoint {
+            public static fromPoint(point: AnnotationPoint): AnnotationMockPoint;
```

Comment:
`point: AnnotationPoint`: shouldn't be `point: Highcharts.Point` ? This static methods creates a new MockPoint from `series.points[index]` point.

Reply:
`AnnotationPoint` is only an internal interface to describe the `Point` instance during annotation processing, so that `any` casts can be avoided. Same for `AnnotationChart` and `AnnotationSeries`.

Reply:
Yes, realized that after writing this comment, thanks!